src/test/datascience/notebook/intellisense/semanticTokens.vscode.test.ts

```diff
@@ -91,4 +89,78 @@ suite('DataScience - VSCode semantic token tests', function () {
         assert.ok(tokens, 'No tokens found on second cell');
         assert.equal(tokens.data[0], 1, 'Tokens not correctly offset');
     });
+
+    test('Edit cells in a notebook', async function () {
+        await insertCodeCell('import sys\nprint(sys.executable)\na = 1');
+        await insertCodeCell('\ndef test():\n  print("test")\ntest()');
+        const cell1 = vscodeNotebook.activeNotebookEditor?.document.cellAt(0)!;
+        const cell2 = vscodeNotebook.activeNotebookEditor?.document.cellAt(1)!;
+
+        const editor = window.visibleTextEditors.find((e) => e.document.uri === cell1.document.uri);
+        await editor?.edit((b) => {
+            b.insert(new Position(2, 0), 'up');
+        });
+
+        // Wait for tokens on the first cell (it works with just plain pylance)
+        await waitForCondition(
+            async () => {
+                const promise = commands.executeCommand('vscode.provideDocumentSemanticTokens', cell1.document.uri);
+                const result = (await promise) as any;
+                return result && result.data.length > 0;
+            },
+            defaultNotebookTestTimeout,
+            `Tokens never appear for first cell`,
+            100,
+            true
+        );
+
+        // Then get tokens on second cell. They should start on line 1. If not this
+        // means there's a bug
+        const tokens = (await commands.executeCommand(
+            'vscode.provideDocumentSemanticTokens',
+            cell2.document.uri
+        )) as any;
+        assert.ok(tokens, 'No tokens found on second cell');
```

Comment:
assertions are the same as the previous test, but we made an edit. Can we check that the edit was performed?

Reply:
Possibly. I could verify the the tokens are not the same as the previous tests tokens (because of the edit). Or hardcode the tokens here and see that they match